chatterbot/adapters/plugins/evaluate_mathematically.py

```diff
@@ -0,0 +1,183 @@
+                    else:
+                        string += str( is_chunk_operator ) + ' '
+                else:
+                    string += str( is_chunk_float ) + ' '
+            else:
+                string += str( is_chunk_integer ) + ' '
+
+        return string
+
+
+    def is_float(self, string):
+        """
+        If the string is a float, returns
+        the float of the string. Otherwise,
+        it returns False.
+        """
+
+        try:
+            return decimal.Decimal(string)
+        except decimal.DecimalException:
+            return False
+
+
+    def is_integer(self, string):
+        """
+        If the string is an integer, returns
+        the int of the string. Otherwise,
+        it returns False.
+        """
+
+        if string.isdigit():
+            return int( string )
+        else:
+            return False
+
+
+    def is_operator(self, string):
+        """
+        If the string is an operator, returns
+        said operator. Otherwise, it returns
+        false.
+        """
+
+        if string in "+-/*^\(\)":
+            return string
+        else:
+            return False
+
+
+    def normalize(self, string):
+        """
+        Normalizes input text, reducing errors
+        and improper calculations.
+        """
+
+        # Setting all words to lowercase
+        string = string.lower()
+
+        # Removing punctuation
+        if string.endswith( ('.', '!', '?', ':', ';' ) ):
+            string = string[ : len(string) - 1 ]
+
+        # Removing words
+        string = self.substitute_words( string )
+
+        # Returning normalized text
+        return string
+
+    def load_data( self, language ):
+        """
+        Load language-specific data
+        """
+
+        if language == "english":
+            with open(os.path.join(os.path.dirname(__file__), 'data', "math_words_EN.json")) as data_file:
+                data = json.load(data_file)
+            self.data = data
+
+
+    def substitute_words(self, string):
+        """
+        Substitutes numbers for words.
+        """
+
+        self.load_data( "english" )
+
+        condensed_string = '_'.join( string.split( ' ' ) )
+
+        for word in self.data[ "words" ]:
+            condensed_string = re.sub( '_'.join( word.split( ' ' ) ), self.data[ "words" ][ word ], condensed_string )
+
+        for number in self.data[ "numbers" ]:
+            condensed_string = re.sub( number, str( self.data[ "numbers" ][ number ] ), condensed_string )
+
+        for scale in self.data[ "scales" ]:
+            condensed_string = re.sub( "_" + scale, " " + self.data[ "scales" ][ scale ], condensed_string)
+
+        condensed_string = condensed_string.split( '_' )
+        for chunk_index in range( 0, len( condensed_string ) ):
+            value = ""
+
+            try:
+                value = str( eval( condensed_string[ chunk_index ] ) )
+
+                condensed_string[ chunk_index ] = value
+            except:
```

Comment:
I am catching all exceptions that rise from attempting to evaluate a non-numerical statement. Because I go through all chunks (including those that are not math expressions) I need to catch all exceptions that might be thrown.